src/XMakeTasks/AssemblyDependency/AssemblyFoldersFromConfig/AssemblyFoldersFromConfig.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Build.Shared;
+using ProcessorArchitecture = System.Reflection.ProcessorArchitecture;
+
+namespace Microsoft.Build.Tasks.AssemblyFoldersFromConfig
+{
+    internal class AssemblyFoldersFromConfig : IEnumerable<AssemblyFoldersFromConfigInfo>
+    {
+        /// <summary>
+        /// The list of directory names found from the config file.
+        /// </summary>
+        private readonly List<AssemblyFoldersFromConfigInfo> _directoryNames = new List<AssemblyFoldersFromConfigInfo>();
+
+        /// <summary>
+        /// Construct.
+        /// </summary>
+        /// <param name="targetRuntimeVersion">The runtime version property from the project file.</param>
+        internal AssemblyFoldersFromConfig(string configFile, string targetRuntimeVersion, ProcessorArchitecture targetArchitecture)
+        {
+            if (!FileUtilities.FileExistsNoThrow(configFile))
+            {
+                return;
+            }
+
+            var collection = AssemblyFolderCollection.Load(configFile);
```

Comment:
Since `Load` just uses deserialization, what's the malformed-input experience like? Should we catch and give a nicer error, or is an uncaught exception understandable?